rust/eos-rs/src/api/math.rs

```diff
@@ -0,0 +1,16 @@
+//! Math functions.
+
+use crate::ffi;
+
+/// Computes the Euclidean norm of a two-component integer array, sort of like hypotf(3).
+pub fn euclidean_norm(vec2: &[i32; 2]) -> f32 {
```

Comment:
I'm curious if using Rust's `f32` will cause LLVM to insert its own IEEE floating-point arithmetic functions for ARMv5TE, even though they already exist in the ROM. That'd be an unfortunate waste of space.

Reply:
I think compiler_builtins would generate them, but I have removed all of the float code from the fork of compiler_builtins that I use for eos_rs. Not sure if there's another part of the toolchain that would try to generate float code.

Reply:
I had some conflicting functions generated in the compiler_builtins crate that were float-related, I think there seems to be some sort of standard for soft-float related function names that Rust also follows.

Actually I googled a bit: 
https://gcc.gnu.org/onlinedocs/gccint/Soft-float-library-routines.html
&
https://github.com/rust-lang/compiler-builtins/blob/master/src/float/add.rs#L194

So yeah, seems to be some GCC established standard.

Reply:
Yeah functions like `__addsf3` are also in arm9.bin in the system lib region. It'd be unfortunate to waste space linking these in when variants already exist internally.

Reply:
Can't check right now: Are they in the symbol tables / more importantly the C header files?

If so both gcc and Rust (which doesn't do it anyways since I remove the code from the compiler_builtins fork) shouldn't try to generate the functions.

Reply:
They are, but not under the "real" names. They're listed as:
- `AddFloat` (`__addsf3`)
- `DivideFloat` (`__divsf3`)
- `FloatToDouble` (`__extendsfdf2`)
- `FloatToInt` (`__fixsfsi`)
- `IntToFloat` (`__floatsisf`)
- `UIntToFloat` (`__floatunsisf`)
- `MultiplyFloat` (`__mulsf3`)
- `SubtractFloat` (`__subsf3`)

There's also integer division stuff, since ARMv5TE doesn't have an instruction for those either:
- `DivideInt` (`__divsi3`)
- `DivideUInt` (`__udivsi3`)

Reply:
Could we alias these symbols using their proper names?
... Then again I don't think you can define functions with `__` in C right? Maybe we need to do this on the linker level in cot.

Reply:
I don't think the linker would complain if we added these to a linker script, but I don't know if they would be used correctly either

Reply:
@UsernameFodder @theCapypara Is it okay for you if we address this in a follow-up? I guess it concerns the C part too anyway.

Reply:
Yeah that's fine. Sorry I didn't make it clear, but I wasn't actually expecting this be to addressed at all, I was just commenting out of curiosity.